src/main/java/fr/esgi/al5_2/Tayarim/services/ProprietaireService.java

```diff
@@ -16,141 +16,222 @@
-        if (proprietaireRepository.findFirstByEmail(proprietaireCreationDTO.getEmail()).isPresent()){
-            throw new ProprietaireEmailAlreadyExistException();
-        }
+    String numTel = proprietaireCreationDTO.getNumTel();
+    numTel = numTel.replaceAll(" ", "");
+    if (proprietaireRepository.findFirstByNumTel(numTel).isPresent()) {
+      throw new ProprietaireNumTelAlreadyExistException();
+    }
+    proprietaireCreationDTO.setNumTel(numTel);
+
+    StringBuilder generatedPassword = new StringBuilder(16);
+    String allowedchar = "abcdefghijklmnopqrstuvwxyz0123456789!@#$%&*()_+-=[]?";
+    SecureRandom random = new SecureRandom();
+    for (int i = 0; i < 16; i++) {
+      int index = random.nextInt(allowedchar.length());
+      generatedPassword.append(allowedchar.charAt(index));
+    }
 
-        String numTel = proprietaireCreationDTO.getNumTel();
-        numTel = numTel.replaceAll(" ", "");
-        if (proprietaireRepository.findFirstByNumTel(numTel).isPresent()){
-            throw new ProprietaireNumTelAlreadyExistException();
-        }
-        proprietaireCreationDTO.setNumTel(numTel);
+    System.out.println(generatedPassword); // waiting for SMTP
 
-        String hashedPassword = hashPassword(proprietaireCreationDTO.getMotDePasse());
+    Proprietaire proprietaire = ProprietaireMapper.creationDtoToEntity(proprietaireCreationDTO,
+        hashPassword(generatedPassword.toString()));
+    return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+  }
 
-        proprietaireCreationDTO.setMotDePasse(hashedPassword);
+  /**
+   * Récupère une liste de tous les propriétaires.
+   *
+   * @param isLogement Indique si les logements associés doivent être inclus.
+   * @return Liste de DTOs des propriétaires.
+   */
+  public List<ProprietaireDTO> getProprietaire(boolean isLogement) {
+    return ProprietaireMapper.entityListToDtoList(proprietaireRepository.findAll(), isLogement);
+  }
 
-        Proprietaire proprietaire = ProprietaireMapper.creationDtoToEntity(proprietaireCreationDTO);
+  public ProprietaireDTO getProprietaireById(@NonNull Long id, boolean isLogement) {
 
-        return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+    Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
+    if (optionalProprietaire.isEmpty()) {
+      throw new ProprietaireNotFoundException();
     }
-
-    public List<ProprietaireDTO> getProprietaire(boolean isLogement){
-        return ProprietaireMapper.entityListToDtoList(proprietaireRepository.findAll(), isLogement);
+    return ProprietaireMapper.entityToDto(optionalProprietaire.get(), isLogement);
+  }
+
+  /**
+   * Récupère un propriétaire par son identifiant, avec la possibilité d'inclure les logements
+   * associés.
+   *
+   * @param id         L'identifiant du propriétaire à récupérer.
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused @param tag for 'id'.